File: exercism/haskell/sum-of-multiples/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

anyDivide :: [Int] -> Int -> Bool
anyDivide divisors n =
  any (\d -> n `mod` d == 0) divisors

sumOfMultiples :: [Int] -> Int -> Int
sumOfMultiples divisors limit =
  sum $ filter (anyDivide divisors) [1..limit - 1]

File: exercism/javascript/sum-of-multiples/sum-of-multiples.js
const { range, sum, some } = require('lodash/fp')

const SumOfMultiples = (factors) => ({
  to: (limit) => {
    const multiples = range(1, limit).filter(m => some(d => m % d == 0, factors))
    return sum(multiples)
  }
})

module.exports = SumOfMultiples
